src/main/kotlin/blackjack/Main.kt

```diff
@@ -1,32 +1,51 @@
 package blackjack
 
-import blackjack.domain.BlackJackCards
+import blackjack.domain.BlackJackDealer
+import blackjack.domain.BlackJackDeck
+import blackjack.domain.BlackJackDeckGenerator
+import blackjack.domain.BlackJackGame
 import blackjack.domain.BlackJackPlayer
 import blackjack.domain.BlackJackPlayerCards
-import blackjack.domain.BlackJackPlayers
-import blackjack.view.BlackJackView
+import blackjack.view.BlackJackResultView
+import blackjack.view.BlackJackInputView
 
 fun main() {
-    val blackJackCards = BlackJackCards.getDefaultCards()
-    val playerNames = BlackJackView.getPlayerName()
-    val blackJackPlayers =
-        BlackJackPlayers(
-            playerNames.map {
-                BlackJackPlayer(
-                    it,
-                    BlackJackPlayerCards(mutableListOf(blackJackCards.draw(), blackJackCards.draw())),
-                )
-            },
-        )
-    BlackJackView.drawBlackJackPlayersCards(blackJackPlayers)
+    val blackJackDeck = BlackJackDeckGenerator.getDefaultDeck()
+    val playerNames = BlackJackInputView.getPlayerName()
+    val blackJackGame = getGame(playerNames, blackJackDeck)
+    BlackJackInputView.drawBlackJackPlayersCards(blackJackGame)
+    doGame(blackJackGame, blackJackDeck)
+    BlackJackResultView.drawBlackJackResult(blackJackGame)
+}
 
-    blackJackPlayers.players.forEach {
-        while (it.isDrawPossible()) {
-            if (!BlackJackView.getPlayerDrawCardYn(it)) break
-            it.drawCard(blackJackCards.draw())
-            BlackJackView.drawBlackJackPlayerCards(it)
+private fun doGame(
+    blackJackGame: BlackJackGame,
+    blackJackDeck: BlackJackDeck,
+) {
+    blackJackGame.players.forEach {
+        while (it.isDrawPossible() && BlackJackInputView.getPlayerDrawCardYn(it)) {
+            it.drawCard(blackJackDeck.draw())
+            BlackJackInputView.drawBlackJackPlayerCards(it)
         }
     }
-    BlackJackView.drawBlackJackPlayersCardsWithResult(blackJackPlayers)
-    BlackJackView.drawWinPlayer(blackJackPlayers)
+    while (blackJackGame.dealer.isDrawPossible()) {
+        blackJackGame.dealer.drawCard(blackJackDeck.draw())
+        println("딜러는 16이하라 한장의 카드를 더 받았습니다.")
+    }
+}
+
+private fun getGame(
+    playerNames: List<String>,
+    blackJackDeck: BlackJackDeck,
+): BlackJackGame {
+    val blackJackGame =
+        BlackJackGame(
+            playerNames.map {
+                BlackJackPlayer(it, BlackJackPlayerCards.byDeck(blackJackDeck))
+            },
+            BlackJackDealer(
+                blackJackPlayerCards = BlackJackPlayerCards.byDeck(blackJackDeck),
+            ),
+        )
+    return blackJackGame
```

Comment:
사소하지만 바로 리턴해줄 수 있을 것 같아요 :) 

```suggestion
       return BlackJackGame(
            playerNames.map {
                BlackJackPlayer(it, BlackJackPlayerCards.byDeck(blackJackDeck))
            },
            BlackJackDealer(
                blackJackPlayerCards = BlackJackPlayerCards.byDeck(blackJackDeck),
            ),
        )
```